cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/AppLifecycleHttpHandler.java

```diff
@@ -341,10 +340,12 @@ public void startStopAdapter(HttpRequest request, HttpResponder responder,
         return;
       }
       responder.sendStatus(HttpResponseStatus.OK);
-    } catch (AdapterNotFoundException e) {
+    } catch (NotFoundException e) {
       responder.sendString(HttpResponseStatus.NOT_FOUND, e.getMessage());
     } catch (InvalidAdapterOperationException e) {
       responder.sendString(HttpResponseStatus.CONFLICT, e.getMessage());
+    } catch (SchedulerException e) {
```

Comment:
I am also wondering what we do with other exceptions? Runtime exceptions etc. Shouldn't they also be treated and logged as internal errors?